mlflow/langchain/__init__.py

```diff
@@ -104,6 +105,23 @@ def get_default_conda_env():
     return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
 
 
+def _get_map_of_special_chain_class_name_to_kwargs_name():
+    from langchain.chains import (
+        APIChain,
+        HypotheticalDocumentEmbedder,
+        RetrievalQA,
+        SQLDatabaseChain,
+    )
+
+    return {
+        RetrievalQA.__name__: "retriever",
+        APIChain.__name__: "requests_wrapper",
+        HypotheticalDocumentEmbedder.__name__: "embeddings",
+        SQLDatabaseChain.__name__: "database",
+        RetrieverWrapper.__name__: "retriever",
+    }
```

Comment:
We need the `__name__` as key because we use this dict to look up the `model_type` string here:
https://github.com/mlflow/mlflow/blob/5b4ace66b482616b2ef6f21716455dfed0826b4f/mlflow/langchain/__init__.py#L583

And the `mode_type` string in written here:
https://github.com/mlflow/mlflow/blob/5b4ace66b482616b2ef6f21716455dfed0826b4f/mlflow/langchain/__init__.py#L255-L258

So we cannot really just use the class object.

Reply:
Got it!